app/components/booking/status-filter.tsx

```diff
@@ -17,11 +18,13 @@ export function StatusFilter({
   const disabled = isFormProcessing(navigation.state);
   const [searchParams, setSearchParams] = useSearchParams();
   const status = searchParams.get("status");
+  const { destoryCookieValues } = useCookieDestory();
 
   function handleValueChange(value: string) {
     setSearchParams((prev) => {
       /** If the value is "ALL", we just remove the param */
       if (value === "ALL") {
+        destoryCookieValues(["status"]);
```

Comment:
I am a bit concerned about this approach, because that requires us to never forget to add this when having functions that clear filters and so on.
A suggestion, what if we create our own abstraction of `useSearchParams()` that handles this and makes sure that every time a value is removed from the search params, we also remove it from the cookie.

Reply:
I could not able to figure out a way to do it due to some scenarios.

1. Not able to differentiate if the user has cleared the filters in the asset page or if the user is directly going to the asset page after some navigations. (in both cases search parameters are empty.)

Reply:
If the user has search params saved in the cookie, they wont be empty as they come with the server response and useSearchParams checks on the client so I am not sure what you mean here.